tests/llama_index/test_llama_index_model_export.py

```diff
@@ -244,3 +262,26 @@ def test_chat_engine_dict_raises(model_path, single_index, with_input_example):
         model = mlflow.pyfunc.load_model(model_path)
         with pytest.raises(TypeError, match="unexpected keyword argument"):
             _ = model.predict(payload)
+
+
+@pytest.mark.parametrize("with_input_example", [True, False])
+def test_retriever_engine_predict(single_index, with_input_example):
+    payload = "string"
+    with mlflow.start_run():
+        model_info = mlflow.llama_index.log_model(
+            index=single_index,
+            artifact_path="model",
+            input_example=payload if with_input_example else None,
+            engine_type="retriever",
+        )
+
+    if with_input_example:
+        assert model_info.signature.inputs is not None
+        # TODO: Inferring signature from retriever output fails because the schema
```

Comment:
Can an example of the retriever expected output be added within this comment block for future reference? 
I'm curious where the None is coming from if we're getting a `[Dict]` output? Is one of the entries in the List `None`?

Reply:
@BenWilson2 Ah sorry for the short words, the `None` is in dictionary value. The `NodeWithSource` defines many fields and some of them can be `None`.

```
 'source_nodes': [{'node': {'id_': '8e22729c-3cd6-422a-82c3-98ba97895830',
    'embedding': None,
    'metadata': {'file_path': '/Workspace/Users/yuki.watanabe@databricks.com/MLflow LlamaIndex/data/MLflow Tracking — MLflow 2.14.1 documentation.html',
    ...
```

It's a bit surprising that the signature framework doesn't handle `None` value tho. We can investigate it (bigger problem than llama-index if so).

Reply:
AH gotcha. 
Yes, this is a bug in signature validation then. Let's file a ticket for next sprint that someone can pick up and address prior to the 2.15.0 RC cut :) 
This actually came up in a different issue yesterday as well for a custom pyfunc. We need to fix this.

Reply:
Makes sense, added to the release blocker table for 2.15.0!